infra/gcp/clusters/modules/k8s-infra-gke-cluster/main.tf

```diff
@@ -55,20 +59,141 @@ resource "google_bigquery_dataset" "usage_metering" {
   }
 
   // This restricts deletion of this dataset if there is data in it
-  // IMPORTANT: Should be true on test clusters
+  // unique to prod_usage_metering
+  delete_contents_on_destroy = false
+}
+resource "google_bigquery_dataset" "test_usage_metering" {
+  count       = var.is_prod_cluster == "true" ? 0 : 1
+  dataset_id  = replace("usage_metering_${var.cluster_name}", "-", "_")
+  project     = var.project_name
+  description = "GKE Usage Metering for cluster '${var.cluster_name}'"
+  location    = var.bigquery_location
+
+  access {
+    role          = "OWNER"
+    special_group = "projectOwners"
+  }
+  access {
+    role          = "WRITER"
+    user_by_email = google_service_account.cluster_node_sa.email
+  }
+
+  // This restricts deletion of this dataset if there is data in it
+  // unique to test_usage_metering
   delete_contents_on_destroy = true
 }
 
 // Create GKE cluster, but with no node pools. Node pools can be provisioned below
-resource "google_container_cluster" "cluster" {
+// Workaround from https://github.com/hashicorp/terraform/issues/22544#issuecomment-582974372
+// to set lifecycle.prevent_destroy to false if is_prod_cluster
+// keep prod_ and test_ identical except for "unique to " comments
+resource "google_container_cluster" "prod_cluster" {
+  count     = var.is_prod_cluster == "true" ? 1 : 0
```

Comment:
What happens if you toggle this on the same object?

Reply:
I think going from not->prod with a bare `terraform apply` would nuke the test resources and create the prod resources. 

I think going from prod->not would leave you with two copies of the resources actuated, even if terraform "forgot" about the prod resources.

I suspect you could get out of this and just have terraform start treating a resource differently if you used `terraform state mv` + `terraform plan` to verify there were no changes to actuate